.github/workflows/rust.yml

```diff
@@ -0,0 +1,22 @@
+name: test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build
```

Comment:
This step won't succeed since the base ubuntu image does not have the rust toolchain installed. 
for that you need to either A) use a different image which includes rust or B) add a step to install rust: 

```
      - name: Install Rust
        uses: actions-rs/toolchain@v1
        with:
          profile: minimal
          toolchain: stable
          override: true
          components: clippy

      - name: Check examples
        uses: actions-rs/cargo@v1
        with:
          command: clippy
          args: --examples --all
```

etc

Reply:
I think that the build is running OK, but maybe I am not getting something.
https://github.com/lambdaclass/cleopatra_cairo/actions